### PR TITLE
Build docker containers for testnet

### PR DESCRIPTION
**Pull request by killerwhile**
This PR adds user.conf for testnet inside the containers.
It also refactor a little bit the Dockerfile to follow the best practices, mostly: group RUN statement to limit layers, reorder to facilitate caching, prefer COPY over ADD.

### CODE REVIEW

**Comment by tdroxler** [comment]
you'll need to run `scalafmtSbt`

small tips, I have the following in my `~/.sbtrc`

```
alias call=;compile;test:compile;it:compile
alias fmt=;scalafmt;test:scalafmt;scalafmtSbt;it:scalafmt
alias style=;scalastyle;test:scalastyle;it:scalastyle
```

File: docker/user-testnet.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 alephium.network.network-type = "testnet"
-alephium.discovery.bootstrap = ["18.221.19.228:9973", "3.142.153.145:9973"]
+alephium.discovery.bootstrap = ["3.68.2.201:9973", "34.236.121.90:9973", "54.252.31.241:9973"]
```
Comment (tdroxler) [inline docker/user-testnet.conf:2]: can't wait the mainnet so we don't need to change that anymore :)

**Review by tdroxler — APPROVED** [review]
LGTM :+1:

**Review by polarker — APPROVED** [review]

File: docker/user-testnet.conf (hunk 1 above)
Comment (killerwhile) [reply docker/user-testnet.conf:2]: We'll introduce DNS soon to mitigate this pain point.

**Comment by killerwhile** [comment]
Thanks for the hint on `~/.sbtrc`!

**Pull request merged**